### DISCUSSION TITLE
Filtering when using transient or computed fields.

### DISCUSSION THREAD

**Discussion by dangoldmanvt1** [intersystems/isc-rest#18] [Q&A] [answered]
In my persistent classes I have quite a few properties like
Property TemplateID As %Integer [ SqlColumnNumber = 2, SqlComputeCode = {Set {*} = {%%ID} }, SqlComputed, Transient ];
I am not able to use these properties in filters like
/templates?templateID[eq]=1 
because 
{
    "errors": [
        {
            "error": "ERROR #5001: Invalid query. Column 'templateID' does not exist.",
            "code": 5001,
            "domain": "%ObjectErrors",
            "id": "GeneralError",
            "params": [
                "Invalid query. Column 'templateID' does not exist."
            ]
        }
    ],
    "summary": "ERROR #5001: Invalid query. Column 'templateID' does not exist."
}

I am trying to figure out a way to use calculated fields and still be able to use them in filtering my rest api.

Thanks
Dan

**Comment by isc-tleavitt** [comment, marked as answer]
@dangoldmanvt1 I think this is an issue in isc.rest (and/or possibly isc.json). We filter out transient properties for query purposes because they normally aren't projected to SQL, but turns out they are if you make them SqlComputed. (I just learned this... today!)

Is there a reason that you Transient rather than Calculated for such fields? I believe Calculated + SqlComputed fields are available for filtering under the current implementation.

**Reply by dangoldmanvt1** [reply]
I was making these transient properties because I do not need them to be stored in database. I am still trying to learn the nuances of Transient, Calculated, SqlComputed ...
Thanks for the solution as it solves my issue!
